docs/contributing/naming-convention.md

```diff
@@ -0,0 +1,27 @@
+# Contributing Naming Convention
+
+## General Guidelines
+
+This document outlines the naming conventions for our project. These conventions are intended to promote consistency and clarity across different components.
+
+### Specific Conventions
+
+#### Python Code
+
+  **Style**: We follow the official Python Enhancement Proposal 8 [PEP8]((https://peps.python.org/pep-0008/#package-and-module-names) ) guidelines for naming conventions.
+
+#### Jupyter Notebooks
+
+ **File Names**: File names should start with an underscore (_). Please use underscores instead of hyphens.
+
+#### Documentation
+
+ **Directory and File Names**: Directory and file names should use hyphens (-) instead of underscores.
+
+#### Tests
+
+1. **Test File and Function Names**: Test files and functions should start with `test_`.
```

Comment:
```suggestion
1. **Test file and member names**: Test files and methods should start with `test_`. Classes should start with `Test`.

    !!! note "Test File Naming Convention"
        We do not use the _test.py suffix for test files

```

Is "member" the right terminology? :shrug: 

I'd like this note to be associated with list item number 1, but I'm not sure if this will format correctly, can you test it out?